xradar/src/main/java/com/orzangleli/radar/XRadarView.java

```diff
@@ -541,38 +616,47 @@ private void drawRegionWithColor(Canvas canvas, float scale) {
             int x, y;
             x = (int) (centerX + scale * percents[i] * radius * Math.cos(angle * i + Math.PI / 2));
             y = (int) (centerY - scale * percents[i] * radius * Math.sin(angle * i + Math.PI / 2));
-            Point p = new Point();
-            p.set(x, y);
+            Point p = new Point(x, y);
             list.add(p);
         }
 
         Path path = new Path();
         for (int i = 0; i < list.size() - 1; i++) {
             path.reset();
             path.moveTo(centerX, centerY);
-            path.lineTo(list.get(i).x, list.get(i).y);
-            path.lineTo(list.get(i + 1).x, list.get(i + 1).y);
+            path.lineTo(list.get(i).getPointX(), list.get(i).getPointY());
+            path.lineTo(list.get(i + 1).getPointX(), list.get(i + 1).getPointY());
             path.lineTo(centerX, centerY);
             path.close();
-            dataPaint.setColor(colors[i % colorSize]);
+            dataPaint.setColor(new Color(colors[i % colorSize]));
             canvas.drawPath(path, dataPaint);
         }
         path.reset();
         path.moveTo(centerX, centerY);
-        path.lineTo(list.get(list.size() - 1).x, list.get(list.size() - 1).y);
-        path.lineTo(list.get(0).x, list.get(0).y);
+        path.lineTo(list.get(list.size() - 1).getPointX(), list.get(list.size() - 1).getPointY());
+        path.lineTo(list.get(0).getPointX(), list.get(0).getPointY());
         path.lineTo(centerX, centerY);
         path.close();
-        dataPaint.setColor(colors[(list.size() - 1) % colorSize]);
+        dataPaint.setColor(new Color(colors[(list.size() - 1) % colorSize]));
         canvas.drawPath(path, dataPaint);
         canvas.restore();
     }
 
-    // 画 多行文字
-    public void drawMultiLinesTextAndIcon(Canvas canvas, float x, float y, CharSequence text, int drawable, int verticalValue, int position) {
+    /**
+     * Draw multiple lines of text
+     *
+     * @param canvas        instance of Canvas
+     * @param x             x coordinate
+     * @param y             y coordinate
+     * @param text          instance of CharSequence to draw text
+     * @param drawable      icon drawable
+     * @param verticalValue vertical value
+     * @param position      position
+     */
+    private void drawMultiLinesTextAndIcon(Canvas canvas, float x, float y, CharSequence text, int drawable, int verticalValue, int position) {
         int drawableAvaiable = 1;
         try {
-            this.getContext().getResources().openRawResource(drawable);
+            this.getContext().getResourceManager().getResource(drawable);
             drawableAvaiable = 1;
```

Comment:
No use of this

Reply:
`this.getContext().getResourceManager().getResource(drawable);`

**Use of this line:**
if this line throw exception then, drawableAvailable = 0 else drawableAvailable = 1